hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/kafka/FilteringMessageReceiver.java

```diff
@@ -31,23 +31,20 @@ public FilteringMessageReceiver(MessageReceiver receiver,
     }
 
     @Override
-    public Message next() {
-        Message message;
-        do {
-            message = receiver.next();
-        } while (consuming && filter(message));
```

Comment:
this method was spinning - now it only reads one and returns optional on each call